lib/_emerge/depgraph.py

```diff
@@ -7370,11 +7370,25 @@ def _wrapped_select_pkg_highest_available_imp(
                     ):
                         continue
 
+                    # We can choose not to install a live package from using binary
+                    # cache by disabling it with option --usepkg-exclude-live in the
+                    # emerge call.
+
+                    usepkg_exclude_live = "--usepkg-exclude-live" in self._frozen_config.myopts
```

Comment:
Since this variable doesn't change inside the loop, we should define it before the loop, next to the other definitions like these:
```
        usepkg = "--usepkg" in self._frozen_config.myopts
        usepkgonly = "--usepkgonly" in self._frozen_config.myopts
```